metric-publishers/emf-metric-publisher/src/main/java/software/amazon/awssdk/metrics/publishers/emf/MetricEmfConverter.java

```diff
@@ -0,0 +1,368 @@
+                Object metricValue = r.value();
+
+                // Store dimension and metric name for later use in Metrics array
+                if (isDimension(metricName)) {
+                    realDimensionStrings.add(metricName);
+                }
+
+                // Add value to aggregated metrics
+                if (shouldReport(r) || isDimension(metricName)) {
+                    aggregatedMetrics.computeIfAbsent(metricName, k -> new ArrayList<>())
+                                     .add(metricValue);
+                }
+
+            });
+
+            // Add children to queue
+            if (current.children() != null) {
+                queue.addAll(current.children());
+            }
+        }
+
+        return createEmfStrings(aggregatedMetrics);
+    }
+
+    private List<String> createEmfStrings(Map<String, List<Object>> aggregatedMetrics) {
+        List<String> emfStrings = new ArrayList<>();
+        Map<String, List<Object>> currentMetricBatch = new HashMap<>();
+        Set<String> currentMetricNames = new HashSet<>();
+
+        // Process metric names and their values
+        for (Map.Entry<String, List<Object>> entry : aggregatedMetrics.entrySet()) {
+            String metricName = entry.getKey();
+            List<Object> values = entry.getValue();
+
+            // Drop large value arrays into chunks of 100
+            if (values.size() > 100) {
+                values = values.subList(0, 100);
+                logger.warn(() -> "Some AWS SDK client-side metric data have been dropped because it exceeds the cloudwatch "
+                                 + "requirements.");
+            }
+            // If adding this metric would exceed 100 metrics, create new batch
+            if (currentMetricNames.size() >= 100) {
+                emfStrings.add(createEmfString(currentMetricBatch, currentMetricNames));
+                currentMetricBatch = new HashMap<>();
+                currentMetricNames = new HashSet<>();
+            }
+
+            currentMetricBatch.put(metricName, values);
+            if (!(values.get(0) instanceof String)) {
+                currentMetricNames.add(metricName);
+            }
+
+        }
+
+        emfStrings.add(createEmfString(currentMetricBatch, currentMetricNames));
+
+        return emfStrings;
+    }
+
+
+    private String createEmfString(Map<String, List<Object>> metrics, Set<String> metricNames) {
+
+        try {
+            JsonWriter jsonWriter = JsonWriter.create();
+            jsonWriter.writeStartObject();
+
+            writeAwsObject(jsonWriter, metricNames);
+            // Write metric values
+            writeMetricValues(jsonWriter, metrics);
+
+            jsonWriter.writeEndObject(); // End root object
+
+            return new String(jsonWriter.getBytes(), StandardCharsets.UTF_8);
+
+        } catch (SdkClientException e) {
+            logger.error(() -> "Failed to create EMF format string");
```

Comment:
https://github.com/aws/aws-sdk-java-v2/blob/master/test/architecture-tests/src/test/java/software/amazon/awssdk/archtests/CodingConventionWithSuppressionTest.java#L80 this test should fail :)  

logger.error is not necessary here because we are throwing an exception anyway